app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -104,13 +106,30 @@ public void onAttach(Context context) {
     }
 
     @Override
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+    public void onCreateOptionsMenu(final Menu menu, MenuInflater inflater) {
         switch (settingsScreen) {
             case SEARCH_ENGINES:
                 inflater.inflate(R.menu.menu_search_engines, menu);
                 break;
             case REMOVE_ENGINES:
                 inflater.inflate(R.menu.menu_remove_search_engines, menu);
+
+                // We disable the trash icon when no engine is checked, enable it otherwise
+                getView().post(new Runnable() {
+                    @Override
+                    public void run() {
+                        final MultiselectSearchEngineListPreference pref = (MultiselectSearchEngineListPreference) getPreferenceScreen()
+                                .findPreference(getResources()
+                                        .getString(R.string.pref_key_multiselect_search_engine_list));
+
+                        if (!pref.isSearchEnginesBindingToMenu()) {
+                            pref.bindEngineCheckboxesToMenu();
```

Comment:
Since this no longer has any dependency on the Menu, and we only want to call it once, we can move this to the MultiselectPreference itself :)